packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -444,13 +443,22 @@ export class StateMachine extends StateMachineBase {
 
     this.stateMachineType = props.stateMachineType ?? StateMachineType.STANDARD;
 
+    let graph: StateGraph | undefined = undefined;
+    if (definitionBody instanceof ChainDefinitionBody) {
+      graph = new StateGraph(definitionBody.chainable.startState, 'State Machine definition');
+      graph.timeout = props.timeout;
+      for (const statement of graph.policyStatements) {
+        this.addToRolePolicy(statement);
```

Comment:
@abdelnn is it possible that this is adding the role policies on the wrong place?

Looking at what it was doing before, I would say that it should be
```typescript
this.role.addToRolePolicy(statement);
```

https://github.com/aws/aws-cdk/issues/29203

Reply:
As soon as I remove this for and put it back where it was before (Only the for).
The bug disappears.

Reply:
I believe you're right, this should be 

`this.role.addToPrincipalPolicy(statement)`